### ISSUE TITLE
Setting breakpoints in realgud:gdb always asks for file location

### ISSUE THREAD

**Issue by topiolli** [realgud/realgud#254]
Realgud version: 20190603-2157
Emacs version: GNU Emacs 26.2 (build 2, x86_64-pc-linux-gnu)

Steps to reproduce:

- Start realgud:gdb and run the program.
- Attach the cmd buffer to a source buffer, let's say this source buffer visits `path/to/src/code.cc`.
- Press b. This prints `break /absolute/path/to/src/code.cc` to the cmd buffer. The path is correct, but because the debugger isn't stopped, nothing else happens. (Is there a way around this?)
- Stop the debugger.
- Press b again. Again, the correct path is printed, and gdb sets the breakpoint. Realgud however asks "Black-list file src/code.cc for location tracking? (y or n)". Pressing "n" brings in another prompt: "Find this source-code file in (default src/code.cc):" I need to type in "/absolute/path/to" to make it happy.
- Answering "y" to the black-list question prints "Wrong type argument: stringp, nil" to the minibuffer.

It seems to me that `realgud:file-loc-from-line` get passed just part of the path although the correct absolute path is known.

**Comment by rocky** [comment]
> This prints break /absolute/path/to/src/code.cc to the cmd buffer. The path is correct, but because the debugger isn't stopped, nothing else happens. (Is there a way around this?)

You'd have to teach `realgud` a way to understand whether gdb is running and paused (that is, looking to read commands) or running and not reading commands. I suppose if you see the `gdb` prompt that might indicate it is waiting on input. 

I am surprised that gdb doesn't queue commands in input. Perhaps however you stop gdb is flushing input. Realgud is just automating the interaction that would happen on the command line. So if you can get this to work on the command line we can probably replicated it in realgud.

> Realgud however asks "Black-list file src/code.cc for location tracking? (y or n)". 

The reason it prompts is because gdb reports back `src/code.cc` as the name of the breakpoint location even though it was passed an absolute file path. 

> Pressing "n" brings in another prompt: "Find this source-code file in (default src/code.cc):" I need to type in "/absolute/path/to" to make it happy.

Admittedly this is lame since this the breakpoint was issued from inside realgud and when it was issued it knew the file path name. 

There are several ways to make gdb smarter, we just need someone who is willing to work on it. The basic idea though is to add a call-back function which will turn the relative path into an absolute path.

See https://github.com/realgud/realgud-jdb/blob/master/realgud-jdb/init.el#L27-L28 and https://github.com/realgud/realgud-jdb/blob/master/realgud-jdb/core.el#L59-L108 for a model for how this works. 

As for ""Wrong type argument: stringp, nil" to the minibuffer." I note that you are not using the latest version of realgud which  is realgud-20190709.2227 . If this is still a problem in the latest version, then run this from git without compiled sources and get and report a traceback using uncompiled sources. 

`M-x toggle-debug-on-entry` will give a traceback when you hit an error.

**Comment by FelipeLema** [comment]
can we quickly send a command and get its output from gdb at this point (inside this would-be `realgud:gdb-find-file`. A solution would to this predicament would be to ask gdb directly for the full path using commands, but I don't know if doing so would mess up eventually (or whether gdb io is block at this stage).

**Comment by rocky** [comment]
> can we quickly send a command and get its output from gdb at this point (inside this would-be realgud:gdb-find-file?. 

Yes, I think that is a possibility. `info file` is verbose, but it should contain information desired. In the command buffer there is a buffer-local variable `divert-output` that is intended to be used to get informtion from the debugger to realgud. See`buffer/common/commnd.el
Inside `realgud/track.el` there where the information is picked up. 

Of course, some experimentation is needed. 

> but I don't know if doing so would mess up eventually (or whether gdb io is block at this stage).

Currently this occasionally happens. I wouldn't worry about it initially.
